cdisc_rules_engine/services/define_xml/base_define_xml_reader.py

```diff
@@ -108,10 +111,27 @@ def extract_variables_metadata(self, domain_name: str = None) -> List[dict]:
                     self._get_item_def_representation(itemdef[0], itemref, codelist_map)
                 )
         logger.info(f"Extracted variables metadata = {variables_metadata}")
+
         return variables_metadata
 
+    @cached("define-metadata-version")
+    def get_metadata_version(self, **kwargs):
+        validate_xml = kwargs.get("validate_xml", "Y")
+        metadata = None
+        try:
+            metadata = self._odm_loader.MetaDataVersion()
+            logger.info(f"Define Metadata Version: {metadata}")
+        except Exception as e:
+            logger._exception = e
+            logger.error(f"{__name__}(VX={validate_xml})")
+            if validate_xml in ("Y", "YES"):
```

Comment:
This line appears a lot across multiple files. Can we at least get it to be a method and call the method instead of having this check in multiple places